### PR TITLE
chore: refactoring mandatory plugin v2

### PR DESCRIPTION
**Pull request by Ash-exp**
# Description

Fixes #

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: util/sliceUtil/SliceUtil.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package sliceUtil
```
Comment (gireesh-naidu) [inline util/sliceUtil/SliceUtil.go:1]: we need to add test cases for these functions

Comment (gireesh-naidu) [reply util/sliceUtil/SliceUtil.go:1]: these functions seems useful as we do these operations at multiple places. we will start using these . so proper test cases are required

File: pkg/plugin/repository/GlobalPluginRepository.go [hunk 2]
```diff
@@ -881,11 +905,18 @@ func (impl *GlobalPluginRepositoryImpl) GetPluginParentMetadataByIds(ids []int)
 }
 
 func (impl *GlobalPluginRepositoryImpl) GetAllPluginMinData() ([]*PluginParentMetadata, error) {
+	return impl.GetAllPluginMinDataByType("")
+}
+
+func (impl *GlobalPluginRepositoryImpl) GetAllPluginMinDataByType(pluginType string) ([]*PluginParentMetadata, error) {
```
Comment (gireesh-naidu) [inline pkg/plugin/repository/GlobalPluginRepository.go:911]: `pluginType` shouldn't be just string , this method should accept the proper type.

Comment (Ash-exp) [reply pkg/plugin/repository/GlobalPluginRepository.go:911]: The method `GetAllPluginMinData()` is being called in multiple places internally. And I introduced `GetAllPluginMinDataByType(pluginType string)` recently for a list API. In the`GetAllPluginMinData()` we want all types of plugins. In order to reduce the impact area and testing scope, This implementation is being placed

File: util/sliceUtil/SliceUtil.go (hunk 1 above)
Comment (Ash-exp) [reply util/sliceUtil/SliceUtil.go:1]: added

**Review by gireesh-naidu — APPROVED** [review]

**Review by nishant-d — APPROVED** [review]

**Pull request merged**